Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -58,34 +58,24 @@ protected function __construct(string $identifier)
     /**
      * Apply an event to the current aggregate root
      *
-     * If the event aggregate identifier and name is not set the event
-     * is automatically set with the current aggregate identifier
-     * and name.
+     * If the event is an instance of AggregateEventInterface, the aggregate identifier will be set
      *
      * @param EventInterface $event
-     * @param array $metadata
-     * @return void
      * @api
      */
-    final public function recordThat(EventInterface $event, array $metadata = [])
+    final public function recordThat(EventInterface $event)
     {
         if ($event instanceof AggregateEventInterface) {
             $event->setIdentifier($this->getIdentifier());
         }
-
-        $messageMetadata = new MessageMetadata($metadata);
-
         $this->apply($event);
-
-        $this->events[] = new EventTransport($event, $messageMetadata);
+        $this->events[] = $event;
```

Comment:
I always consider the version of the aggregate to be an internal invariant of the Aggregate itself, which basically states how many events it already has applied to itself. Since the version should always be consistent with the current state of the aggregate, it needs to be updated when applying the event.

The problem I see is that from the point the aggregate gets some events applied to the point it is actually persisted (eventually), its version and hence the aggregate is inconsistent.